Spatial_Engine/Query/IsFullyOnLevel.cs

```diff
@@ -0,0 +1,117 @@
+        /******************************************/
+
+        [Description("Checks if the geometrical representation of an IElement0D is fully within a set distance from a level elevation.")]
+        [Input("element0D", "The IElement0D that will be checked for proximity to the Level.")]
+        [Input("level", "The Level to use for evaulation.")]
+        [Input("maxDistance", "The maximum distance allowed from the Level for this method to return true.", typeof(Length))]
+        [Output("isFullyOnLevel", "A boolean which is true if the geometrical representation of the IElement0D is fully within a set distance from the level elevation.")]
+        public static bool IsFullyOnLevel(this IElement0D element0D, Level level, double maxDistance)
+        {
+            Point position = element0D.IGeometry();
+
+            return Math.Abs(level.Elevation - position.Z) <= maxDistance;
+        }
+
+        /******************************************/
+
+        [Description("Checks if the geometrical representation of an IElement1D is fully within a set distance from a level elevation.")]
+        [Input("element1D", "The IElement1D that will be checked for proximity to the Level.")]
+        [Input("level", "The Level to use for evaulation.")]
+        [Input("maxDistance", "The maximum distance allowed from the Level elevation for this method to return true.", typeof(Length))]
+        [Output("isFullyOnLevel", "A boolean which is true if the geometrical representation of an IElement1D is fully within a set distance from the level elevation.")]
+        public static bool IsFullyOnLevel(this IElement1D element1D, Level level, double maxDistance)
+        {
+            ICurve curve = element1D.IGeometry();
+            BoundingBox bBox = curve.IBounds();
+
+            return level.Elevation - maxDistance <= bBox.Min.Z && bBox.Max.Z <= level.Elevation + maxDistance;
+        }
+
+        /******************************************/
+
+        [Description("Checks if the geometrical representation of an IElement2D is fully within a set distance from a level elevation.")]
+        [Input("element2D", "The IElement2D that will be checked for proximity to the Level.")]
+        [Input("level", "The Level to use for evaulation.")]
+        [Input("maxDistance", "The maximum distance allowed from the Level for this method to return true.", typeof(Length))]
+        [Output("isFullyOnLevel", "A boolean which is true if the geometrical representation of an IElement2D is fully within a set distance from the level elevation.")]
+        public static bool IsFullyOnLevel(this IElement2D element2D, Level level, double maxDistance)
+        {
+            List<IElement1D> elements1D = element2D.IOutlineElements1D();
+
+            bool isFullyOnLevel = true;
+
+            foreach (IElement1D e1D in elements1D)
+            {
+                isFullyOnLevel = isFullyOnLevel && IsFullyOnLevel(e1D, level, maxDistance);
```

Comment:
```suggestion
                isFullyOnLevel &= IsFullyOnLevel(e1D, level, maxDistance);
```

Reply:
Nice, did not know you could do that! Makes sense though since you can do it with the mathematical operators